src/main/webapp/WEB-INF/tags/instructor/results/questionPanel.tag

```diff
@@ -45,18 +45,18 @@
     </div>
     <div <c:if test="${questionPanel.collapsible}">class="${questionPanel.responsesBodyClass}"</c:if>>
         <div class="panel-body padding-0" <c:if test="${questionIndex != null}">id="questionBody-${questionIndex}"</c:if>>
-            
-            <c:if test="${!questionPanel.questionHasResponses}">
+            <c:if test="${(empty questionPanel.responses && questionPanel.showResponseRows) || !isShowingAll}">
```

Comment:
We'd better use 
'''
<c:choose>
     <c:when test="${(empty questionPanel.responses && questionPanel.showResponseRows) || !isShowingAll}">
           .......................
     </c:when>
     <c:otherwise>
          .............
     </c:otherwise>
'''
here to make sure it doesn't show both the message and the table and to simplify the logic as well.